packages/react-aria-components/docs/Table.mdx

```diff
@@ -999,6 +999,471 @@ Use the `renderEmptyState` prop to customize what the `TableBody` will display i
 </Table>
 ```
 
+## Column Resizing
+
+Table supports resizable columns, allowing users to dynamically adjust the width of a column. This is enabled by wrapping the `<Table>` with a `<ResizableTableContainer>` element, which serves as a scrollable container for the Table. Then, to make a column resizable, render a `<ColumnResizer>` element inside a `<Column>`. This allows a user to drag a resize handle to change the width of a column. Keyboard users can also resize a column by pressing <Keyboard>Enter</Keyboard> to enter resizing mode and then using the arrow keys to resize. Screen reader users can resize columns by operating the resizer like a slider.
+
+### Width values
+
+By default, a Table relies on the browser's default table layout algorithm to determine column widths. However, when a Table is wrapped in a `<ResizableTableContainer>`, column widths are calculated in JavaScript by React Aria. When no additional props are provided, Table divides the available space evenly among the columns. The `Column` component also supports four different width props that allow you to control column sizing behavior: `defaultWidth`, `width`, `minWidth`, and `maxWidth`.
+
+An initial, uncontrolled width can be provided to a Column using the `defaultWidth` prop. This allows the column width to freely shrink and grow in relation to other column widths. Alternatively, a controlled value can be provided by the `width` prop. These props accept fixed pixel values, percentages of the total table width, or [fractional](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Basic_Concepts_of_Grid_Layout#the_fr_unit) values (the `fr` unit), which represent a fraction of the available space. Columns without a defined width are equivalent to `1fr`.
+
+The `minWidth` and `maxWidth` props define constraints on the size of a column, which may be defined either as fixed pixel values or as percentages of the total table width. These are respected when calculating the size of a column, and also provide limits for column resizing.
+
+The example below illustrates how each of the column width props affects their respective column's resize behavior.
+
+```tsx example
+import {ResizableTableContainer, ColumnResizer} from 'react-aria-components';
+
+<ResizableTableContainer>
+  <Table aria-label="Table with resizable columns">
+    <TableHeader>
+      {/*- begin highlight -*/}
+      <Column id="file" isRowHeader maxWidth={500}>
+        <div className="flex-wrapper">
+          <span tabIndex={-1} className="column-name">File Name</span>
```

Comment:
Not sure about this. The tabIndex is so that the name can be focused, but idk if this needs a role or something. It seemed to announce ok in VoiceOver (as a group). Not sure if we should explain this somewhere, or even provide a component for this?

Reply:
Not sure if it needs a role either, it just announced the span text in TalkBack without a role. I think we should explain it at the very least in passing or as a comment at the very least. Making it a component might be a bit overkill, but I could see an argument for making one if we need it for other situations as well.

Reply:
cc @majornista is it valid to have a focusable element without a role?